test-framework/vertx/src/main/java/io/quarkus/test/vertx/RunOnVertxContextTestMethodInvoker.java

```diff
@@ -74,7 +75,7 @@ public Object invoke(Object actualTestInstance, Method actualTestMethod, List<Ob
         CompletableFuture<Object> cf = new CompletableFuture<>();
         RunTestMethodOnContextHandler handler = new RunTestMethodOnContextHandler(actualTestInstance, actualTestMethod,
                 actualTestMethodArgs, uniAsserter, cf);
-        vertx.getOrCreateContext().runOnContext(handler);
+        VertxContext.getOrCreateDuplicatedContext(vertx).runOnContext(handler);
```

Comment:
I think we need to give more flexibility. Yes, the default should be a duplicated context with its root being the event loop context, but we may need some other cases.

That being said, that should always run on a duplicated context.